doc/api/perf_hooks.md

```diff
@@ -521,6 +521,22 @@ property will be an {Object} with two properties:
   * `perf_hooks.constants.NODE_PERFORMANCE_GC_FLAGS_ALL_EXTERNAL_MEMORY`
   * `perf_hooks.constants.NODE_PERFORMANCE_GC_FLAGS_SCHEDULE_IDLE`
 
+### HTTP ('http') Details
+
+When `performanceEntry.type` is equal to `'http'`, the
+`performanceEntry.detail` property will be an {Object} containing
+additional information.
+
+If `performanceEntry.name` is equal to `HttpClient`, the `detail`
+will contain the following properties: `req`, `res`. And the `req` property
+will be an {Object} containing `method`, `url`, `headers`, the `res` property
+will be an {Object} containing `statusCode`, `statusMessage`, `headers`.
+
+If `performanceEntry.name` is equal to `HttpRequest`, the `detail`
+will contain the following properties: `req`, `res`. And the `req` property
+will be an {Object} containing `method`, `url`, `headers`, the `res` property
+will be an {Object} containing `statusCode`, `statusMessage`, `headers`.
```

Comment:
I question if including the `headers` is a good idea. The performance entry information ends up being aggregated and deferred to after the request. This would have the effect of forcing the headers to be held on to longer than they otherwise likely would have been and would mean the headers are available outside of the context of the request to which they are relevant.

Reply:
Thanks for review! Do this(the headers are available outside of the context of the request to which they are relevant) have a bad effect ? What i think is the `headers` is useful for users to analyze which request is slow.

Reply:
> I question if including the `headers` is a good idea. The performance entry information ends up being aggregated and deferred to after the request. This would have the effect of forcing the headers to be held on to longer than they otherwise likely would have been and would mean the headers are available outside of the context of the request to which they are relevant.

Thanks for review! Do this(the headers are available outside of the context of the request to which they are relevant) have a bad effect ? What i think is the headers is useful for users to analyze which request is slow.

Reply:
> I agree. Usually you should not use this by default on production systems. If you're enabling it you're probably temporary troubleshooting problems and thus this transient overhead is probably fine.

Thanks. I remove the `headers` in this PR.

Reply:
@theanarkh  Sorry, I think I misexplained. I was agreeing with you, headers will not harm and should be present.

Reply:
Oh, I add it back and see what other people think. Thanks !

Reply:
> @theanarkh Sorry, I think I misexplained. I was agreeing with you, headers will not harm and should be present.

Oh, I add it back and see what other people think. Thanks !

Reply:
> @theanarkh Sorry, I think I misexplained. I was agreeing with you, headers will not harm and should be present.

Oh, I add it back and waiting for other opinion. Thanks !

Reply:
Ok. I would add a comment in the docs then that this could add additional memory overhead and should likely only be used for diagnostic purposes and not just left turned on in production.

Reply:
Good idea. I agree!

Reply:
Sorry, i have updated the docs.